apps/aetx/src/aetx.erl

```diff
@@ -154,19 +157,25 @@
     Tx :: tx().
 new(Callback, Tx) ->
     Type = Callback:type(),
-    #aetx{ type = Type, cb = Callback, tx = Tx }.
+    {Vsn, Fields} = Callback:serialize(Tx),
+    Template = Callback:serialization_template(Vsn),
+    Size = byte_size(aec_object_serialization:serialize(Type, Vsn, Template, Fields)),
+    #aetx{ type = Type, cb = Callback, size = Size, tx = Tx }.
 
 -spec fee(Tx :: tx()) -> Fee :: integer().
 fee(#aetx{ cb = CB, tx = Tx }) ->
     CB:fee(Tx).
 
 -spec gas(Tx :: tx()) -> Gas :: non_neg_integer().
-gas(#aetx{ cb = CB, tx = Tx }) ->
-    CB:gas(Tx).
+gas(#aetx{ cb = CB, size = Size, tx = Tx }) ->
+    CB:gas(Tx, Size).
```

Comment:
I would prefer: `CB:gas(Tx) + Size * aec_governance:byte_gas()`

Reply:
@hanssv @jur0 I kind of liked it when I run into it. In the future we may distinguish "dummy size" and size affecting state trees. E.g. Spend payload are dead bytes, but all the hashes in tx fields require computation etc. So the only place having context about it, would be tx module.